src/components/auth/torus/sdk/TorusSDK.js

```diff
@@ -16,8 +16,8 @@ import {
 class TorusSDK {
   strategies = {}
 
-  static factory() {
-    const sdk = new TorusSDK(Config, logger.child({ from: 'TorusSDK' }))
+  static factory(configOverride = {}) {
+    const sdk = new TorusSDK({ ...Config, ...configOverride }, logger.child({ from: 'TorusSDK' }))
```

Comment:
```suggestion
  static factory(customUxMode = null) {
    const sdk = new TorusSDK(Config, logger.child({ from: 'TorusSDK' }), customUxMode)
```

lines 32-34
```javascript
  constructor(config, logger, customUxMode = null) {
    const { env, publicUrl, torusProxyContract, torusNetwork, torusUxMode } = config
    const uxMode = customUxMode || torusUxMode || 'popup'
    const popupMode = uxMode === 'popup'
```

line 42
```
uxMode,
```

Reply:
yes in this case i override uxmode, maybe in the future we want to override other stuff

Reply:
overriding an app config object. also isn't good idea.
pls look at this solution:

```javascript
import { defaults } from 'lodash'

// options now are TorusSDK options, not a config override
static factory(options = null) {
  const sdk = new TorusSDK(Config, options, logger.child({ from: 'TorusSDK' }))
  
  // ...
} 

// ...
constructor(config, options, logger) {
    const { env, publicUrl, torusProxyContract, torusNetwork, torusUxMode = 'popup' } = config
    // using options passed firstly, append with values from the app config for the opts aren't overridden
    const torusOptions = defaults({}, options, {
      proxyContractAddress: torusProxyContract, // details for test net
      network: torusNetwork, // details for test net
      enableLogging: env === 'development',
      uxMode: torusUxMode
    })

    const popupMode = torusOptions.uxMode === 'popup'
    const baseUrl = publicUrl + (popupMode ? '/torus' : '')
    const redirectPath = popupMode ? 'redirect' : 'Welcome/Auth'

    // setting values for url & redirect if  aren't overridden
    // doing this separately as we need to determine uxMode firstly
    defaults(torusOptions, { baseUrl, redirectPathName: redirectPath })

    this.torus = new Torus(config, torusOptions)
    assign(this, { popupMode, config, logger })
  }
```

Reply:
usage:

```javascript
setSDK(TorusSDK.factory({ uxMode: torusUxMode }))
```

Reply:
ok